components/ai_chat/core/browser/engine/engine_consumer.cc

```diff
@@ -20,7 +20,7 @@ namespace ai_chat {
 
 namespace {
 
-constexpr char kArrayPattern[] = R"((\[.*?\]))";
+constexpr char kArrayPattern[] = R"((?s)(\[.*?\]))";
```

Comment:
what changed that we have to handle newlines in v2?

Reply:
During my testing I encountered server sending me topics array with '\n' inside which failed my parsing, I'm not sure why we don't see it in v1, or it actually happens in v1 too sometimes just we haven't noticed before.
AFAIK, I think v1 would also fail if it receive this kind of request because the updated case of including '\n' in the response in v1 engine consumer would fail without this change too.